javaslang/src/main/java/javaslang/collection/AbstractsQueue.java

```diff
@@ -0,0 +1,100 @@
+package javaslang.collection;
+
+import javaslang.*;
+import javaslang.control.Option;
+
+import java.util.*;
+
+abstract class AbstractsQueue<T, Q extends AbstractsQueue<T, Q>> implements Kind1<Q, T>, Traversable<T> {
```

Comment:
Let's implement the `Kind` not in the abstract class. It is some sort of terminating resistor. They should be implemented by `Queue` and `PriorityQueue`. Also we should make these classes final because some methods overloaded from `Traversable` resp. `LinearSeq` are terminated by types, i.e. the methods are not overloadable by other classes any more.